backend/cluster/cluster_crud_test.go

```diff
@@ -0,0 +1,101 @@
+package cluster
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/ZPI-2024-25/KubernetesAccessManager/models"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+)
+
+type MockResourceInterface struct {
+	dynamic.ResourceInterface
+	mock.Mock
+	ReturnedGetValue *unstructured.Unstructured
+	ReturnedGetError error
+}
+
+func (m *MockResourceInterface) Get(ctx context.Context, name string, options metav1.GetOptions, subresources ...string) (*unstructured.Unstructured, error) {
+	return m.ReturnedGetValue, m.ReturnedGetError
+}
+
+func TestGetResourceError(t *testing.T) {
+
+	mockGetResourceI := new(mock.Mock)
+	mockGetResourceI.On("func1", "Pod", "default", "default").Return(&MockResourceInterface{}, &models.ModelError{Code: 404, Message: "Not found"})
+	getResourceI = func(resourceType string, namespace string, emptyNamespace string) (dynamic.ResourceInterface, *models.ModelError) {
+		args := mockGetResourceI.Called(resourceType, namespace, emptyNamespace)
+		return args.Get(0).(dynamic.ResourceInterface), args.Get(1).(*models.ModelError)
+	}
+
+	t.Run("Test GetResource", func(t *testing.T) {
+		_, err := GetResource("Pod", "default", "default")
+		assert.NotNil(t, err)
```

Comment:
Change it to verify if received error is exactly what we want - message, code, and is this an object that we want, not sth else

Reply:
General comment

Reply:
resolved, I should be checking all error values now